nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-connection.js

```diff
@@ -1820,6 +1820,41 @@
                 .attrs({
                     'filter': getDropShadow
                 });
+
+            // penalized icon
+            var connectionLabelContainer = updated.select('g.connection-label-container');
+            if (connectionLabelContainer.select('text.penalized-icon').empty()) {
+                connectionLabelContainer.select('g.queued-container')
+                    .append('text')
+                    .attrs({
+                        'class': 'penalized-icon',
+                        'y': 14
+                    })
+                    .text(function () {
+                        return '\uf252';
+                    })
+                    .append('title');
+            }
+
+            // determine whether or not to show the penalized icon
+            connectionLabelContainer.select('text.penalized-icon')
+                .classed('hidden', function () {
+                    var flowFileAvailability = _.get(d, 'status.aggregateSnapshot.flowFileAvailability', null)
```

Comment:
The callback to `classed` provides `d` as an argument. In the `updateConnections` function, we were in a for each context that already provided it. In the `updateConnectionStatus` we are not and its need and needs to be parameterized in the callback.

> nf-connection.js?1.16.0-SNAPSHOT:1842 Uncaught ReferenceError: d is not defined
    at SVGTextElement.<anonymous> (nf-connection.js?1.16.0-SNAPSHOT:1842:54)